### ISSUE TITLE
Specify swiftformat version in your .swiftformat file?

### ISSUE THREAD

**Issue by simondelphia** [nicklockwood/SwiftFormat#908]
Is there a way to specify a SwiftFormat version in the `.swiftformat` file so that when you run `.swiftformat` and your local version of SwiftFormat does not match the version in the rules file it errors out and suggests updating? 

You can easily have multiple people on one repo using different versions of SwiftFormat and then you get different formatting results depending on who is running the formatter and it's not immediately obvious why.

**Comment by nicklockwood** [comment]
@simondelphia yes! You can specify a minimum SwiftFormat version using `--minversion`

**Comment by simondelphia** [comment]
Is there a way to specify the version exactly? If I put `--minversion 0.48` someone running `0.47` won't get an error

**Comment by nicklockwood** [comment]
@simondelphia no, that's not possible at the moment. I could add a --maxversion or --exactversion parameter as well.

**Comment by simondelphia** [comment]
Could it actually result in different formatting results? Or does SwiftFormat follow the rules according to the lower version?

**Comment by nicklockwood** [comment]
SwiftFormat doesn't do anything to change its behavior according to --minversion, although that's not a bad idea - I could disable new rules that were introduced after the --minversion automatically for example.

I remember now that one of the reasons I didn't add --maxversion is that it's relatively difficult to downgrade to an earlier version, so it's probably better to update the --minversion to the latest release instead.

**Comment by simondelphia** [comment]
Well maybe at least a warning saying `your minversion specified 0.48 but you are using 0.49` or something?

**Comment by H-G-Hristov** [comment]
It will be nice it there is an option to support `minswiftformat` version, `minswiftversion` or just `swiftversion` in `.swiftformat`.  Also probably document the proper number format. Invalid versions doesn't seem to report an error like if I put `5.445343` in `.swift-version` doesn't seem to output an error.

**Comment by nicklockwood** [comment]
@H-G-Hristov AFAIK both `--minversion` and `--swiftversion` are both supported in `.swiftformat` config. Are they not working for you?

The version format is also enforced, but `5.445343` isn't an error because that's a valid format under the semver pattern (there doesn't happen to be a version like that, but there might be one day).

**Comment by c0diq** [comment]
I specified `--minversion 0.48.2` in the config file. Yet one developer with an older version (0.47.13) is choking with unknown option `--emptybraces` which is defined before that and continues trying to lint. Can we force the tool to error immediately even if `--lenient` is specified?

**Comment by nicklockwood** [comment]
@c0diq can you share the config file you're using? I'll try to reproduce the issue.

**Comment by kanishkajay** [comment]
Is there any way to downgrade swiftformat?

**Comment by nicklockwood** [comment]
You can download old builds from the [releases page](https://github.com/nicklockwood/SwiftFormat/tags).
